### PR TITLE
Compound like Governance Contracts

### LINKED ISSUE #48: Dojo Voting Module
This task is to create a voting module based on Dojo Models to allow games to implement voting strategies using Dojo models. For the first version, we should simply demonstrate how to create a voting structure with a basic strategy.

The structure should include:

Primitives (This should be based within a new Governance directory):
- [ ]  Submission of the transaction to execute if the strategy passes quorum. This could be stored in a generic Model structure.
- [ ]  A mechanism to execute the proposed transaction if it passes the strategy.
- [ ]  Definition of the generic strategy and initialization of it. For example, does it require ERC721 votes, ERC20 votes, or a custom Entity within a Dojo world to vote?

Examples:
- [ ]  An example demonstrating the usage of the generic voting primitives based on ERC721 Votes.

Inspiration for this module could be found within the open-zeppelin contracts.

### PR DESCRIPTION
**Pull request by ametel01**
Closes #48 

## Introduced changes
Introducing a governance mechanism inspired to [Compound Governance](https://github.com/compound-finance/compound-protocol/tree/master/contracts/Governance), [docs](https://docs.compound.finance/v2/governance/)  

The contracts are added as a workspace member of the repo.

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Add a dedicated CI job for new examples
- [x] Performed self-review of the code

### CODE REVIEW

File: governance/src/systems/token/interface.cairo [hunk 1]
```diff
@@ -0,0 +1,21 @@
+use starknet::ContractAddress;
+
+#[dojo::interface]
+trait IGovernanceToken {
+    fn spaw(
```
Comment (ponderingdemocritus) [inline governance/src/systems/token/interface.cairo:5]: spawn?

Comment (ametel01) [reply governance/src/systems/token/interface.cairo:5]: token constructor

**Comment by ametel01** [comment]
Contracts completed, i'll start with testing

**Comment by ponderingdemocritus** [comment]
Could you include an example using this crate - thank you!

**Comment by ametel01** [comment]
> Could you include an example using this crate - thank you!

This is supposed to be a draft or, still missing some testing. I'll make it draft for now

**Pull request merged**